cms/djangoapps/contentstore/views/tests/test_discussion_enabled.py

```diff
@@ -0,0 +1,119 @@
+        )
+        self.vertical = ItemFactory(
+            parent_location=self.sequential.location,
+            category="vertical",
+            display_name="Triskelion",
+            modulestore=self.store
+        )
+        self.vertical_1 = ItemFactory(
+            parent_location=self.sequential.location,
+            category="vertical",
+            display_name="Helicarrier",
+            modulestore=self.store
+        )
+        self.course.save()
+        return self.course
+
+    def _get_discussion_enabled_status(self, usage_key, client=None):
+        """
+        Issue a GET request to fetch value of discussion_enabled flag of xblock represented by param:usage_key
+        """
+        client = client if client is not None else self.client
+        url = reverse_usage_url("xblock_handler", usage_key)
+        resp = client.get(url, HTTP_ACCEPT="application/json")
+        content = json.loads(resp.content.decode("utf-8"))
+        return content.get("discussion_enabled", None)
+
+    def get_discussion_enabled_status(self, xblock, client=None):
+        """
+        Issue a GET request to fetch value of discussion_enabled flag of param:xblock's
+        """
+        return self._get_discussion_enabled_status(xblock.location, client=client)
+
+    def set_discussion_enabled_status(self, xblock, value, client=None):
+        """
+        Issue a POST request to update value of discussion_enabled flag of param:xblock's
+        """
+        client = client if client is not None else self.client
+        xblock_location = xblock.location
+        url = reverse_usage_url("xblock_handler", xblock_location)
+        resp = client.post(
+            url,
+            HTTP_ACCEPT="application/json",
+            data=json.dumps({"metadata": {"discussion_enabled": value}}),
+            content_type="application/json",
+        )
+        return resp
+
+    def test_discussion_enabled_false_initially(self):
+        """
+        Tests discussion_enabled flag is False initially for vertical
+        """
+        self.assertFalse(self.get_discussion_enabled_status(self.vertical))
+        self.assertFalse(self.get_discussion_enabled_status(self.vertical_1))
+
+    def test_discussion_enabled_toggle(self):
+        """
+        Tests discussion_enabled can be toggled.
+        """
+        self.set_discussion_enabled_status(self.vertical, True)
+        self.assertTrue(self.get_discussion_enabled_status(self.vertical))
+        self.assertFalse(self.get_discussion_enabled_status(self.vertical_1))
+
+    def test_non_course_author_cannot_get_or_set_discussion_enabled_flag(self):
+        """
+        Test non course author cannot get/set discussion_enabled flag
+        """
+        with self.assertRaises(json.JSONDecodeError):
+            self.assertEqual(
+                self._get_discussion_enabled_status(self.course_usage_key, self.non_staff_authed_user_client),
+                "disabled"
+            )
```

Comment:
@farhaanbukhsh ^

Reply:
Hey @awaisdar001  sorry for the delay I will have a look and reply to as soon as possible :)

Reply:
Hey @awaisdar001 we are catching the JSONDecodeError because when a non-staff user makes the request we are sending back 

```
b'\n<!doctype html>\n<html lang="en">\n<head>\n  <title>403 Forbidden</title>\n</head>\n<body>\n  <h1>403 Forbidden</h1><p></p>\n</body>\n</html>\n'
```

which isn't a proper JSON. Thinking about it, this doesn't seems like a clean way to achieve the test. So I am pushing a refined way of achieving the same thing.